etl/steps/data/meadow/oecd/2025-02-25/social_expenditure.py

```diff
@@ -0,0 +1,128 @@
+    "TP51": {"new_name": "Total", "category": "Family"},
+    "TP511": {"new_name": "Family allowances", "category": "Family"},
+    "TP512": {"new_name": "Maternity and parental leave", "category": "Family"},
+    "TP513": {"new_name": "Other cash benefits", "category": "Family"},
+    "TP521": {"new_name": "Early childhood education and care (ECEC)", "category": "Family"},
+    "TP522": {"new_name": "Home help and accommodation", "category": "Family"},
+    "TP523": {"new_name": "Other benefits in kind", "category": "Family"},
+    "TP60": {"new_name": "Total", "category": "Active labor market programs"},
+    "TP601": {"new_name": "PES and Administration", "category": "Active labor market programs"},
+    "TP602": {"new_name": "Training", "category": "Active labor market programs"},
+    "TP603": {"new_name": "Job rotation and job sharing", "category": "Active labor market programs"},
+    "TP604": {"new_name": "Employment incentives", "category": "Active labor market programs"},
+    "TP605": {"new_name": "Supported employment and rehabilitation", "category": "Active labor market programs"},
+    "TP606": {"new_name": "Direct job creation", "category": "Active labor market programs"},
+    "TP607": {"new_name": "Start-up incentives", "category": "Active labor market programs"},
+    "TP71": {"new_name": "Total", "category": "Unemployment"},
+    "TP711": {"new_name": "Unemployment compensation and severance pay", "category": "Unemployment"},
+    "TP712": {"new_name": "Early retirement for labor market reasons", "category": "Unemployment"},
+    "TP82": {"new_name": "Total", "category": "Housing"},
+    "TP821": {"new_name": "Housing assistance", "category": "Housing"},
+    "TP822": {"new_name": "Other benefits in kind", "category": "Housing"},
+    "TP91": {"new_name": "Total", "category": "Other social policy areas"},
+    "TP911": {"new_name": "Income maintenance", "category": "Other social policy areas"},
+    "TP912": {"new_name": "Other cash benefits", "category": "Other social policy areas"},
+    "TP921": {"new_name": "Social assistance", "category": "Other social policy areas"},
+    "TP922": {"new_name": "Other benefits in kind", "category": "Other social policy areas"},
+}
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("social_expenditure.csv")
+
+    # Load data from snapshot.
+    tb = snap.read()
+
+    #
+    # Process data.
+    #
+    # Keep only the columns of interest.
+    tb = tb[COLUMNS_TO_KEEP.keys()]
+
+    # Rename columns.
+    tb = tb.rename(columns=COLUMNS_TO_KEEP)
+
+    # Map indicator codes to their descriptions and create a new column called programme_type_category
+    for code, mapping in PROGRAMME_TYPE_CODES.items():
+        tb.loc[tb["programme_type"] == code, "programme_type_category"] = mapping["category"]
+        tb.loc[tb["programme_type"] == code, "programme_type"] = mapping["new_name"]
```

Comment:
I think you could avoid the for with something like (i haven't tested):

```python
tb["programme_type_category"] = tb["programme_type"].apply(lambda x: x["category"])
tb["programme_type"] = tb["programme_type"].apply(lambda x: x["new_name"])
```

Reply:
Thanks, I applied
```
tb["programme_type_category"] = tb["programme_type"].map(lambda x: PROGRAMME_TYPE_CODES[x]["category"])
tb["programme_type"] = tb["programme_type"].map(lambda x: PROGRAMME_TYPE_CODES[x]["new_name"])
```